create_faiss_rag_db.py

```
@@ -0,0 +1,657 @@
+
+    def _clone_repo(self):
+        """Clone the OpenShift Lightspeed RAG content repository"""
+        try:
+            subprocess.run(
+                ["git", "clone", OPENSHIFT_LIGHTSPEED_REPO, str(self.repo_dir)],
+                check=True,
+                capture_output=True,
+                text=True,
+            )
+        except subprocess.CalledProcessError as e:
+            raise RuntimeError(f"Failed to clone repository: {e.stderr}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)